frontend/src/views/Lobby.tsx

```diff
@@ -279,6 +288,9 @@ function LobbyPage() {
     if (!loading) {
       setLoading(true);
       setSpectator(currentRoomId, request)
+        .then((res) => {
+          setStateFromRoom(res);
```

Comment:
> also, since set spectator sends a socket update, is there any need for this then block to set the room? if you do tho, I think you can replace this `setStateFromRoom` call with `dispatch(setRoom(res))`, and then in `setStateFromRoom` you should be able to remove the dispatch set room there. 

That makes sense; in that case, I don't think I'd need to add `dispatch(setUser(newUserWithSpectateValue))` as you suggested, I could just remove this whole `then` block. I'll try that and see if it still updates.

Reply:
I just removed the whole `then` block, and the room still updates correctly 👍